src/main/java/com/madgag/gif/fmsware/AnimatedGifEncoder.java

```diff
@@ -345,6 +372,30 @@ protected int findClosest(Color c) {
 	}
 	
 	/**
+	 * Returns index of palette color closest to c
+	 *
+	 */
+	protected int findExact (Color c) {
+		if (colorTab == null) return -1;
+		int r = c.getRed() & 0xff;
+		int g = c.getGreen() & 0xff;
+		int b = c.getBlue() & 0xff;
+		int len = colorTab.length;
+		for (int i = 0; i < len; i++) {
+			int index = i / 3;
+			if (usedEntry[index]) {
+				int tr = colorTab[i++] & 0xff;
+				int tg = colorTab[i++] & 0xff;
+				int tb = colorTab[i] & 0xff;
+				if (r==tr && g==tg && b==tb) {

```

Comment:
Here it looks like, for one iteration of the loop, `i` is incremented by `3` if `usedEntry[index] == true`, but otherwise **`1`** - because 2 of the increments only occur in the `if` statement.

That seems likely to be wrong, comparing this with `findClosest()`?